package/MDAnalysis/auxiliary/__init__.py

```diff
@@ -0,0 +1,519 @@
+  # will return the default XVGReader class
+  auxreader = MDAnalysis.auxiliary.code.get_auxreader_for(format='XVG-F')
+  # will return the XVGFileReader class
+
+To directly load an instance of the guessed auxiliary reader class given the
+supplied auxdata and any additional auxiliary reader options, the 
+function :func:`~MDAnalysis.auxiliary.core.auxreader` can be used::
+
+  aux = MDAnalysis.auxiliary.auxreader('auxdata.xvg', dt=2)
+  for auxstep in aux:
+     do_something(auxstep)
+
+
+Auxiliaries and trajectories
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For alignment with trajectory data, auxiliary readers provide methods to 
+assign each auxiliary step to the nearest trajectory timestep, read all steps
+assigned to a trajectory timestep and calculate 'representative' value(s) of 
+the auxiliary data for that timestep. 
+
+'Assignment' of auxiliary steps to trajectory timesteps is determined from the time 
+of the auxiliary step, ``dt`` of the trajectory and time at the first frame of the 
+trajectory (obtained through a :class:`~MDAnalysis.coordinates.base.Timestep` 
+instance from the trajectory), as::
+
+    frame = floor((time_at_step - time_at_frame_0 + dt/2)/dt)
+
+If there are no auxiliary steps assigned to a given timestep (or none within 
+``cutoff``, if set), the representative value(s) are set to ``np.nan``.
+
+Adding an auxiliary to a trajectory
+...................................
+Auxiliary data may be added to a trajectory 
+(:class:`~MDAnalysis.coordinates.base.Reader` object) through the 
+:meth:`~MDAnalysis.coordinates.base.ProtoReader.add_auxiliary` method. Auxiliary data
+may be passed in as a AuxReader instance, or directly as e.g. a filename, in 
+which case :func:`~MDAnalysis.auxiliary.core.get_auxreader_for` is used to 
+guess an appropriate reader. e.g.::
+
+    u = MDAnalysis.Universe(PDB, XTC)
+    u.trajectory.add_auxiliary('pullforce', './pull_force.xvg')
+
+As the trajectory frame is updated, the auxiliary data will be read 
+correspondingly, 
+updated, and the representative auxiliary value(s) will be available as e.g.
+``u.trajectory.ts.aux.pullforce``.
+
+
+Iterating by an auxiliary
+.........................
+The trajectory :class:`~MDAnalysis.coordinates.base.Reader` methods 
+:meth:`~MDAnalysis.coordinates.base.ProtoReader.next_as_aux` and
+:meth:`~MDAnalysis.coordinates.base.ProtoReader.iter_as_aux` allow for movement
+through only trajectory timesteps to which one or more steps that fall within 
+``cutoff`` from a given auxiliary have been assigned. This may be used to 
+avoid representative values set to ``np.nan``, particularly when auxiliary data
+is less frequent::
+
+    u.trajectory.add_auxiliary('low_f', 'low_freq_aux_data.xvg')
+    for ts in u.iter_as_aux():
+        do_something(ts.aux.low_f) # do something with auxiliary data without
+                                   # worrying about having to deal with np.nan
+
+If the auxiliary data are more frequent and if set the cutoff (if set) is 
```

Comment:
I'd accidentally left an extra 'if set' in there; hopefully it reads clearer now?